File: c_language_solutions/move_zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int j = 0;
    int num[numsSize]; // Creating a new array to store non-zero elements

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            num[j] = nums[i];
            j++;
        }
    }

    // Fill remaining elements of num with zeroes
    for (int i = j; i < numsSize; i++) {
        num[i] = 0;
    }

    // Copy elements of num back to nums
    for (int i = 0; i < numsSize; i++) {
        nums[i] = num[i];
    }
}

File: python_codes/move_zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        non_zero_index = 0
        
        # Move all non-zero elements to the beginning of the list
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[non_zero_index] = nums[i]
                non_zero_index += 1
        
        # Fill the rest of the list with zeros
        for i in range(non_zero_index, len(nums)):
            nums[i] = 0
